src/libcmd/built-path.cc

```diff
@@ -58,6 +58,21 @@ StorePathSet BuiltPath::outPaths() const
     );
 }
 
+StorePaths BuiltPath::outPaths2() const
+{
+    return std::visit(
+        overloaded{
+            [](const BuiltPath::Opaque & p) { return StorePaths{p.path}; },
+            [](const BuiltPath::Built & b) {
+                StorePaths res;
+                for (auto & [_, path] : b.outputs)
+                    res.push_back(path);
+                return res;
+            },
+        }, raw()
+    );
+}
```

Comment:
Maybe not, but it made implementing easier, since I didn't have to figure out how to convert a `std::set` to a `std::vector`. I'll do some more playing around.